src/NonCumulative/Soundness/Properties/LogRel.agda

```diff
@@ -86,4 +88,620 @@ Glu-wellfounded-≡-GluΛ {j} {k} = implicit-extensionality fext (fext (λ l<k 
+             Γ ⊢ t ∶ T ®[ i ] a ∈El B≈A
+  ®El-swap {_} {_} {Γ} {t} {T} {i = i} (ne′ C≈C′) (ne C′≈C _ _) (ne c≈c refl _ , glu) = (ne c≈c refl refl) , record
+    { t∶T  = t∶T
+    ; ⊢T   = ⊢T
+    ; krip = λ ⊢σ → krip′ ⊢σ
+    }
+    where
+      open GluNe glu
+      krip′ : Δ ⊢w σ ∶ Γ →
+               -----------------------------------
+               Δ ⊢ T [ σ ] ≈ Ne⇒Exp (proj₁ (C′≈C (L.length Δ))) ∶[ 1 + i ] Se i
+                 × Δ ⊢ t [ σ ] ≈ Ne⇒Exp (proj₁ (c≈c (L.length Δ))) ∶[ i ] T [ σ ]
+      krip′ {Δ} {σ} ⊢σ
+        with C≈C′ (len Δ) | C′≈C (len Δ) | krip ⊢σ
+      ...  | _ , ↘u , _ | _ , _ , ↘u₁ | Tσ≈ , tσ≈
+           rewrite Re-det ↘u ↘u₁ = Tσ≈ , tσ≈
+  ®El-swap N′ N′ T® = T®
+  ®El-swap U′ (U i≡1+j j≡j′) T®
+    rewrite ≡-irrelevant i≡1+j refl = T®
+  ®El-swap (Π′ {j} {k} jA RT) (Π i≡maxjk jA′ RT′ j≡j′ k≡k′) record { t∶T = t∶T ; a∈El = a∈El ; IT = IT ; OT = OT ; ⊢IT = ⊢IT ; ⊢OT = ⊢OT ; T≈ = T≈ ; krip = krip }
+    rewrite ≡-irrelevant i≡maxjk refl
+          | 𝕌-wf-gen j (ΠI≤′ j k refl)
+          | 𝕌-wf-gen k (ΠO≤′ j k refl)
+          | Glu-wf-gen j (ΠI≤′ j k refl)
+          | Glu-wf-gen k (ΠO≤′ j k refl)
+    with Π≈Π′ , f≈f′∈ElΠ ← Π-bundle jA (λ a≈b → RT a≈b , a∈El a≈b) refl
+          = record
+    { t∶T  = t∶T
+    ; a∈El = El-Π-𝕌 i≡maxjk jA′ RT′ (El-swap Π≈Π′ (Π-𝕌 jA′ RT′ i≡maxjk) f≈f′∈ElΠ)
+    ; IT   = IT
+    ; OT   = OT
+    ; ⊢IT  = ⊢IT
+    ; ⊢OT  = ⊢OT
+    ; T≈   = T≈
+    ; krip = λ ⊢σ → let open ΛRel (krip ⊢σ) in record
+      { IT-rel = ®-swap jA jA′ IT-rel
+      ; ap-rel = λ s® b∈ → ap-helper b∈ s® ap-rel
+      }
+    }
+    where ap-helper : (b∈′ : b ∈′ El j jA′) →
+                      Δ ⊢ s ∶ IT [ σ ] ®[ j ] b ∈El jA′ →
+                      (∀ {s b} → Δ ⊢ s ∶ IT [ σ ] ®[ j ] b ∈El jA →
+                        (a∈ : b ∈′ El j jA) →
+                        ΛKripke Δ (t [ σ ] $ s) (OT [ σ , s ∶ IT ↙ j ]) a b (_⊢_∶_®[ k ]_∈El ΠRT.T≈T′ (RT a∈)) ) →
+                      --------------------------------------------------------------
+                      ΛKripke Δ (t [ σ ] $ s) (OT [ σ , s ∶ IT ↙ j ]) a b (_⊢_∶_®[ k ]_∈El ΠRT.T≈T′ (RT′ b∈′))
+          ap-helper b∈′ s® ap-rel
+             with El-sym jA′ jA b∈′
+          ...  | b∈
+              with RT b∈ | RT′ b∈′ | ap-rel (®El-swap jA′ jA s®) b∈
+          ... | record { ↘⟦T⟧ = ↘⟦T⟧ ; ↘⟦T′⟧ = ↘⟦T′⟧ ; T≈T′ = T≈T′ }
+              | record { ↘⟦T⟧ = ↘⟦T′⟧₁ ; ↘⟦T′⟧ = ↘⟦T⟧₁ ; T≈T′ = T′≈T }
+              | record { fa = fa ; ↘fa = ↘fa ; ®fa = ®fa }
+              rewrite ⟦⟧-det ↘⟦T⟧ ↘⟦T⟧₁
+                    | ⟦⟧-det ↘⟦T′⟧ ↘⟦T′⟧₁ = record { fa = fa ; ↘fa = ↘fa ; ®fa = ®El-swap T≈T′ T′≈T ®fa }
+
+  ®El-swap (L′ {j} {k} kA) (L i≡j+k kA′ j≡j′ k≡k′) record { t∶T = t∶T ; UT = UT ; ⊢UT = ⊢UT ; a∈El = a∈El ; T≈ = T≈ ; krip = krip }
+    rewrite ≡-irrelevant i≡j+k refl
+          | 𝕌-wf-gen {j + k} k (Li≤ refl)
+          | Glu-wf-gen k (Li≤′ j k refl)
+    with Li≈Li′ , a≈a′∈ElLi ← L-bundle {j = j} kA a∈El refl = record
```

Comment:
is the implicit argument necessary?

Reply:
yes, otherwise Agda will complain, though it should guess `j` from `refl`